### ISSUE TITLE
Invalid fix for PLE1141

### ISSUE THREAD

**Issue by martinlehoux** [astral-sh/ruff#13343]
- I think the autofix is not safe
- Ruff version: 0.6.4
- Ruff command: `ruff check --preview --fix`
```py
d: dict[tuple[int, str], bool] = {}
d[1, "a"] = True

for n, c in d:
    print(n, c)
```

autofixed to 

```py
d: dict[tuple[int, str], bool] = {}
d[1, "a"] = True

for n, c in d.items():
    print(n, c)
```

**Comment by MichaReiser** [comment]
To summarize the issue:

* `for n, c in d` iterates over the dictionary keys
* `for n, c in d.items()` iterates over the `key-value` pairs. 

It's questionable if PLE1141 should be raised for your example because the code only iterates over the keys. Pylint had a similar issue a long time ago https://github.com/pylint-dev/pylint/issues/3283

**Comment by martinlehoux** [comment]
Yes exactly. But as I understand it, ruff doesn't use type checking, so it would be difficult to say that the key is a tuple.

**Comment by henribru** [comment]
It seems like the fix for this should at the very least be marked as unsafe, since in general it will be impossible to determine if the key is a tuple, even if support was added for inferring it in obvious cases.
